app/src/main/java/de/danoeh/antennapod/fragment/preferences/SynchronizationPreferencesFragment.java

```diff
@@ -0,0 +1,320 @@
+                        break;
+                    default:
+                        break;
+                }
+                setSelectedSyncProvider(userSelect);
+                if (userSelect.equals(SYNC_PROVIDER_CHOICE_GPODDER_NET)) {
+                    new GpodderAuthenticationFragment()
+                            .show(getChildFragmentManager(), GpodderAuthenticationFragment.TAG);
+                    updateScreen();
+                    return;
+                }
+                openNextcloudAccountChooser();
+                updateScreen();
+            });
+
+            AlertDialog dialog = builder.create();
+            dialog.show();
+            return true;
+        });
+        findPreference(PREF_GPODNET_SETLOGIN_INFORMATION)
+                .setOnPreferenceClickListener(preference -> {
+                    AuthenticationDialog dialog = new AuthenticationDialog(activity,
+                            R.string.pref_gpodnet_setlogin_information_title, false, GpodnetPreferences.getUsername(),
+                            null) {
+
+                        @Override
+                        protected void onConfirmed(String username, String password) {
+                            GpodnetPreferences.setPassword(password);
+                            SyncService.setIsProviderConnected(getContext(), true);
+                        }
+                    };
+                    dialog.show();
+                    return true;
+                });
+        findPreference(PREFERENCE_SYNC).setOnPreferenceClickListener(preference -> {
+            SyncService.syncImmediately(getActivity().getApplicationContext());
+            return true;
+        });
+        findPreference(PREFERENCE_FORCE_FULL_SYNC).setOnPreferenceClickListener(preference -> {
+            SyncService.fullSync(getContext());
+            return true;
+        });
+        findPreference(PREFERENCE_LOGOUT).setOnPreferenceClickListener(preference -> {
+            GpodnetPreferences.logout();
+            Snackbar.make(getView(), R.string.pref_gpodnet_logout_toast, Snackbar.LENGTH_LONG).show();
+            SyncService.setIsProviderConnected(getContext(), false);
+            setSelectedSyncProvider("n/a");
+            updateScreen();
+            return true;
+        });
+    }
+
+    private void setSelectedSyncProvider(String userSelect) {
+        SharedPreferences preferences = getActivity()
+                .getSharedPreferences(SyncService.SHARED_PREFERENCES_SYNCHRONIZATION, Activity.MODE_PRIVATE);
+        preferences.edit()
+                .putString(SyncService.SHARED_PREFERENCE_SELECTED_SYNC_PROVIDER, userSelect)
+                .apply();
+    }
+
+    private void updateScreen() {
+        final boolean loggedIn = SyncService.isProviderConnected(getContext());
+        findPreference(PREFERENCE_LOGIN).setVisible(!loggedIn);
+
+        int icon = getIconForSelectedSyncProvider(getSelectedSyncProviderKey());
```

Comment:
How about adding a method to each sync provider that returns its icon and text? Then the providers are more self contained. Currently, quite many `switch` statements need to be updated when (in the future) adding additional sync providers - might be easy to miss one of them.

Reply:
I tried adding methods (getHeader, getIcon...) to the sync providers. But i does not feel right since the added methods deal with "view stuff" that have no dependency to the sync provider. Also it does not help because the fragment does not know about the sync services. It just knows about user selection (which is a string in shared preferences).

Reply:
What about associative arrays in the preferences where one needs to add icon and text for each sync provider?
Then we can query the icon and text by reading the value by key from that array. Does that sound ok?

Reply:
I extracted the logic for resolving header summary, icon and username to a separate class that gets the values from string array resources or static access objects. This way its more open to extending with other providers or at least encapsulated in a single class
Is that ok?